src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/filter_table.go

```diff
@@ -71,25 +73,44 @@ func (s *filterStatements) prepare(db *sql.DB) (err error) {
 
 func (s *filterStatements) selectFilter(
 	ctx context.Context, localpart string, filterID string,
-) (filter []byte, err error) {
-	err = s.selectFilterStmt.QueryRowContext(ctx, localpart, filterID).Scan(&filter)
-	return
+) (*gomatrix.Filter, error) {
+	// Retrieve filter from database (stored as canonical JSON)
+	var filterData []byte
+	err := s.selectFilterStmt.QueryRowContext(ctx, localpart, filterID).Scan(&filterData)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshall JSON into Filter struct
+	var filter gomatrix.Filter
+	if err = json.Unmarshal(filterData, &filter); err != nil {
+		return nil, err
+	}
+	return &filter, nil
 }
 
 func (s *filterStatements) insertFilter(
-	ctx context.Context, filter []byte, localpart string,
+	ctx context.Context, filter *gomatrix.Filter, localpart string,
```

Comment:
It feels kind of odd to be using a client library to get these types rather than including them in gomatrixserverlib. Seems like we download the entire gomatrix lib just for the `Filter` type. That seems kind of silly.

Probably in a separate PR, but someone needs to go in and include `Filter` structs in gomatrixserverlib and then use those here instead.

Reply:
Sound legit
Should I "copy/paste" [this pr](https://github.com/matrix-org/gomatrix/pull/46) to gomatrixserverlib?

Reply:
Yes, that looks like it should do the trick!

Reply:
=> https://github.com/matrix-org/gomatrixserverlib/pull/90

Reply:
/me now humbly requests a `gomatrix` -> `gomatrixserverlib` switch and `gomatrixserverlib` commit hash update.